analyzers/src/SonarAnalyzer.Common/Rules/ParametersCorrectOrderBase.cs

```diff
@@ -18,141 +18,161 @@
-            MethodParameterLookupBase<TArgumentSyntax> methodParameterLookup, SeparatedSyntaxList<TArgumentSyntax> argumentList,
-            Func<Location> getLocationToReport)
-        {
-            var argumentParameterMappings = methodParameterLookup.GetAllArgumentParameterMappings()
-                .ToDictionary(pair => pair.Node, pair => pair.Symbol);
+    protected abstract TypeInfo GetArgumentTypeSymbolInfo(TArgumentSyntax argument, SemanticModel model);
+    protected abstract Location GetMethodDeclarationIdentifierLocation(SyntaxNode syntaxNode);
+    protected abstract SyntaxToken? GetArgumentIdentifier(TArgumentSyntax argument, SemanticModel model);
+    protected abstract SyntaxToken? GetNameColonArgumentIdentifier(TArgumentSyntax argument);
 
-            var methodSymbol = methodParameterLookup.MethodSymbol;
-            if (methodSymbol == null)
-            {
-                return;
-            }
+    protected ParametersCorrectOrderBase() : base(DiagnosticId) =>
+        rule = Language.CreateDescriptor(DiagnosticId, MessageFormat);
 
-            var parameterNames = argumentParameterMappings.Values
-                .Select(symbol => symbol.Name)
-                .Distinct()
-                .ToList();
-
-            var argumentIdentifiers = argumentList
-                .Select(argument => ConvertToArgumentIdentifier(argument))
-                .ToList();
-            var identifierNames = argumentIdentifiers
-                .Select(p => p.IdentifierName)
-                .ToList();
-
-            if (parameterNames.Intersect(identifierNames).Any() &&
-                HasIncorrectlyOrderedParameters(argumentIdentifiers, argumentParameterMappings, parameterNames, identifierNames,
-                    analysisContext.SemanticModel))
-            {
-                // for VB the symbol does not contain the method syntax reference
-                var secondaryLocations = methodSymbol.DeclaringSyntaxReferences
-                    .Select(s => GetMethodDeclarationIdentifierLocation(s.GetSyntax()))
-                    .WhereNotNull();
+    internal void ReportIncorrectlyOrderedParameters(SonarSyntaxNodeReportingContext analysisContext,
+        MethodParameterLookupBase<TArgumentSyntax> methodParameterLookup,
+        SeparatedSyntaxList<TArgumentSyntax> argumentList,
+        Func<Location> getLocationToReport)
+    {
+        var argumentParameterMappings = methodParameterLookup.GetAllArgumentParameterMappings()
+                                                             .ToDictionary(x => x.Node, x => x.Symbol);
 
-                analysisContext.ReportIssue(SupportedDiagnostics[0].CreateDiagnostic(analysisContext.Compilation,  getLocationToReport(), secondaryLocations, methodSymbol.Name));
-            }
+        var methodSymbol = methodParameterLookup.MethodSymbol;
+        if (methodSymbol == null)
+        {
+            return;
```

Comment:
It's a cheap property and used only twice => no need to extract.
```suggestion
        if (methodSymbol is methodParameterLookup.MethodSymbol)
        {
            return;
```